signature/src/verifier.rs

```diff
@@ -0,0 +1,106 @@
+use crate::suite::{Ed25519Signature, VerificationRelation};
+pub use signature::Signature;
+
+pub struct Ed25519DidVerifier {
+    public_key: ed25519_zebra::VerificationKey,
+}
+
+impl From<&crate::signer::Ed25519DidSigner> for Ed25519DidVerifier {
+    fn from(signer: &crate::signer::Ed25519DidSigner) -> Self {
+        Self {
+            public_key: signer.public_key,
+        }
+    }
+}
+
+pub trait DIDVerifier<S>: signature::Verifier<S>
+where
+    S: signature::Signature,
+{
+    fn decoded_verify(&self, msg: &[u8], data: String) -> Result<(), signature::Error> {
+        let decoded_sig = self.decode(data)?;
+        return self.verify(msg, &decoded_sig);
+    }
+
+    fn decoded_relational_verify(
+        &self,
+        msg: &[u8],
+        data: String,
+        relation: VerificationRelation,
+    ) -> Result<(), signature::Error>;
+
+    fn relational_verify(
+        &self,
+        msg: &[u8],
+        signature: &S,
+        relation: VerificationRelation,
+    ) -> Result<(), signature::Error>;
+
+    fn decode(&self, encoded_sig: String) -> Result<S, signature::Error>;
+}
+
+impl signature::Verifier<Ed25519Signature> for Ed25519DidVerifier {
+    fn verify(&self, msg: &[u8], sig: &Ed25519Signature) -> Result<(), signature::Error> {
+        let sig_bytes: [u8; 64] = sig.0.as_slice().try_into().unwrap();
+        self.public_key
+            .verify(&ed25519_zebra::Signature::from(sig_bytes), msg)
+            .map_err(|_| signature::Error::new())
```

Comment:
@mkatychev Both are from external crates so it doesn't seem like I can do that?

Reply:
Isn't `signature::Error` in SSI?

Reply:
That's actually defined by https://github.com/RustCrypto/traits/tree/master/signature, we have a bit of a weird dependency with them which is why I proposed that we remove that altogether since its not necessary

Reply:
I see, or you can wrap a signature::Error using `thiserror` that way you have control over your methods.

Reply:
I've removed signature, let me know what you think